app/models/teacher.rb

```diff
@@ -3,4 +3,7 @@
 class Teacher < User
   has_many :teacher_classrooms, dependent: :destroy
   has_many :classrooms, through: :teacher_classrooms
+
+  validates :email, presence: true, uniqueness: true, format: { with: URI::MailTo::EMAIL_REGEXP,
```

Comment:
Let's remove the `presence: true` we don't want to require an email.  Admins may create teachers without emails to start with so this is fine.  And Teacher inherits from User and user allow email to be nil so this will cause conflicts.